internal/cmd/builtin/cmd/generate.go

```diff
@@ -255,18 +402,66 @@ func constructValue(v reflect.Value) (jen.Code, error) {
 		typ := v.Type()
 		s := indirectType(typ)
 		values := make(jen.Dict, v.NumField())
-		for i := 0; i < v.NumField(); i++ {
-			field := v.Field(i)
-			if !field.CanInterface() {
-				// Ignore private fields
-				continue
+		if typ.Name() == "DateTimeLiteral" {
```

Comment:
For readability I think it would help to use a switch statement here and then to put all of this logic into its own function that way we do not have so many nested scopes.

Reply:
I agree.